src/resolver.cpp

```diff
@@ -21,103 +27,176 @@ TF_DEBUG_CODES(OPENASSETIO_RESOLVER)
 
 PXR_NAMESPACE_CLOSE_SCOPE
 
+namespace {
+/// Converter logger from OpenAssetIO log framing to USD log outputs.
+class UsdOpenAssetIOResolverLogger : public openassetio::log::LoggerInterface {
+ public:
+  void log(Severity severity, const openassetio::Str &message) override {
+    switch (severity) {
+      case Severity::kCritical:
+        TF_ERROR(TfDiagnosticType::TF_DIAGNOSTIC_FATAL_ERROR_TYPE, message + "\n");
```

Comment:
[fix]/[plan] Pretty sure we discovered this was adding an extra newline, i.e. the `\n` isn't necessary here (and for some of the other severities).

Reply:
Yeah, did my own test and confirmed TF_ERROR and TF_WARN add newlines implicitly.

https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/16/commits/25bb4b99a864b2be5e7e46798cc14ad76d1a92d7